src/stacked_borrows/mod.rs

```diff
@@ -958,7 +924,8 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn retag_reference(
         &mut self,
         val: &ImmTy<'tcx, Provenance>,
-        kind: RefKind,
+        ref_kind: RefKind,
+        retag_kind: RetagKind,
```

Comment:
Please add comments explaining that `retag_kind` is for diagnostic purposes only.